File: DristorApp/Data/Models/Address.cs
using System;
using System.ComponentModel.DataAnnotations
;
namespace DristorApp.Data.Models
{
    public class Address
    {
        public int Id { set; get; }
        public string Country { set; get; }
        public string City { set; get; }
        public string AddressLine { set; get; }
        public string PostalCode { set; get; }
        public string PhoneNumber { set; get; }

        [Required]
        public User User { set; get; }

        public ICollection<Order> Orders { set; get; }

    }
}

File: frontend/src/app/models/address.ts
import {IUser} from "./user";
import {IOrder} from "./order";

export interface IAddress {
  id: number;
  county: string;
  city: string;
  addressLine: string;
  postalCode: string;
  phoneNumber: string;

  user?: IUser;
  order?: IOrder;
}

export interface IAddressCreate{
  county: string;
  city: string;
  addressLine: string;
  postalCode: string;
  phoneNumber: string;

  user: number;
}
export interface IAddressUpdate extends IAddressCreate{ }
